app/src/main/java/org/thoughtcrime/securesms/conversation/newmessage/NewMessageViewModel.kt

```diff
@@ -31,7 +31,7 @@ class NewMessageViewModel @Inject constructor(
     val state = _state.asStateFlow()
 
     private val _event = Channel<Event>()
-    val event = _event.receiveAsFlow()
+    val event: Flow<Event> get() = _event.receiveAsFlow()
```

Comment:
If this is true here, this should be applied everywhere, ie: DisappearingMessages VM for example.

Reply:
Apparently this doesn't actually fix the issue. 

> A `Channel` is a hot stream that can only deliver each item to one receiver.

But the issue also doesn't actually exist, because we don't subscribe to it twice at the same time anyway, and we probably shouldn't.

Probably won't need this kinda thing when we have compose navigation.

Reply:
Wouldn't you subscribe to it a second time on configuration change? when the VM survives but the activity/fragment doesn't?

Reply:
@bemusementpark

Reply:
no, you unsubscribe, then subscribe again.

a Channel emits to one subscriber, so if you only have one, its fine... I think.

Reply:
When I mentioned changing field to getter, I was talking from a perspective of an API expose. If you see a Java API `Flow<Event> getEvent()` you would have slightly more confidence than a field `final Flow<Event> event`, in that the method call may yield a new `Flow` each time, so that you don't try to collect the same `Flow` twice. It's probably just me clinging to the old age of RxJava where whenever I see a `Flow`, I treat it as collectable-only-once. 

That being said, I look up `receiveAsFlow` and it does support more than one collector so the original code is probably working ok. However,  I'd personally use `SharedFlow`: the interface tells you unambiguously that unlike a plain `Flow`, it (`SharedFlow`) can be shared. Like this: 

```kotlin
private val _event = MutableSharedFlow<Event>()
val event: SharedFlow<Event> get() = _event

```

Reply:
cool, maybe:

```
private val _events = MutableSharedFlow<Event>()
val events = _event.asSharedFlow() // hides mutability from consumers.
```

aside: which `receiveAsFlow`?  `ReceiveChannel#receiveAsFlow` says only one of the flows will get emissions, so better to not `get()` on that. but anyway, `Channel` seems to be the wrong class.

Reply:
done